dali/pipeline/operator/operator.h

```diff
@@ -156,8 +156,10 @@ class DLL_PUBLIC OperatorBase {
 
   /**
    * @brief Saves operator state into a checkpoint.
+   *
+   * Is called exactly once per epoch.
   */
-  virtual void SaveState(OpCheckpoint &cpt, std::optional<cudaStream_t> stream) const {
+  virtual void SaveState(OpCheckpoint &cpt, std::optional<cudaStream_t> stream) {
```

Comment:
Actually, can we keep it `const` and not make this assumption? Not that it particularily hurt us, but I just find it a bit surprising for the SaveState method to change state of the operator.
Is that because of the removing the state from the loader's queue? How about removing the stale state from the queue when the state is becomes stale, i.e. in the Run method?

Reply:
The problem I faced at first was that I can't lock a mutex with the `const`, and I don't really know how to deal with it, as the queue is updated from another thread.